src/query/query_executor.c

```diff
@@ -20078,14 +20068,23 @@ qexec_analytic_eval_instnum_pred (THREAD_ENTRY * thread_p, ANALYTIC_STATE * anal
       return NO_ERROR;
     }
 
+  analytic_state->xasl->instnum_flag &= ~(XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC);
   /* evaluate inst_num() */
   is_output_rec = qexec_eval_instnum_pred (thread_p, analytic_state->xasl, analytic_state->xasl_state);
   if (is_output_rec == V_ERROR)
     {
+      if (instnum_flag & XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC)
+	{
+	  analytic_state->xasl->instnum_flag |= XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC;
+	}
       return ER_FAILED;
     }
   else
     {
+      if (instnum_flag & XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC)
+	{
+	  analytic_state->xasl->instnum_flag |= XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC;
+	}
       analytic_state->is_output_rec = (is_output_rec == V_TRUE);
     }
```

Comment:
same code in if-else block. please review again.
```suggestion
if (instnum_flag & XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC)
  {
    analytic_state->xasl->instnum_flag |= XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC;
  }
if (is_output_rec == V_ERROR)
  {
    return ER_FAILED;
  }
else
  {
    analytic_state->is_output_rec = (is_output_rec == V_TRUE);
  }
```

Reply:
fixed it